test/sample-test/sample_test_launcher.py

```diff
@@ -97,29 +94,72 @@ def _compile_sample(self):
+      if m:
+        file_name, ext_name = os.path.splitext(file)
+        if self._is_notebook is not None:
+          raise(RuntimeError('Multiple entry points found under sample: {}'.format(self._test_name)))
+        if ext_name == '.py':
+          self._is_notebook = False
+        if ext_name == '.ipynb':
+          self._is_notebook = True
+
+    if self._is_notebook is None:
+      raise(RuntimeError('No entry point found for sample: {}'.format(self._test_name)))
+
+    config_schema = yamale.make_schema(SCHEMA_CONFIG)
+    # Retrieve default config
+    try:
+      with open(DEFAULT_CONFIG, 'r') as f:
+        raw_args = yaml.safe_load(f)
+      default_config = yamale.make_data(DEFAULT_CONFIG)
+      yamale.validate(config_schema, default_config)  # If fails, a ValueError will be raised.
+    except yaml.YAMLError as yamlerr:
+      raise RuntimeError('Illegal default config:{}'.format(yamlerr))
+    except OSError as ose:
+      raise FileExistsError('Default config not found:{}'.format(ose))
+    else:
+      self._run_pipeline = raw_args['run_pipeline']
+
     # For presubmit check, do not do any image injection as for now.
     # Notebook samples need to be papermilled first.
-    if self._test_name == 'lightweight_component':
-      pm.execute_notebook(
-          input_path='Lightweight Python components - basics.ipynb',
-          output_path='%s.ipynb' % self._test_name,
-          parameters=dict(
-              EXPERIMENT_NAME='%s-test' % self._test_name
-          )
-      )
-    elif self._test_name == 'dsl_static_type_checking':
+    if self._is_notebook:
+      # Parse necessary params from config.yaml
+      nb_params = {}
+
+      try:
+        with open(os.path.join(CONFIG_DIR, '%s.config.yaml' % self._test_name), 'r') as f:
+          raw_args = yaml.safe_load(f)
+        test_config = yamale.make_data(os.path.join(
+          CONFIG_DIR, '%s.config.yaml' % self._test_name))
+        yamale.validate(config_schema, test_config)  # If fails, a ValueError will be raised.
+      except yaml.YAMLError as yamlerr:
+        print('No legit yaml config file found, use default args:{}'.format(yamlerr))
+      except OSError as ose:
+        print('Config file with the same name not found, use default args:{}'.format(ose))
+      else:
+        if 'notebook_params' in raw_args.keys():
+          nb_params.update(raw_args['notebook_params'])
+        if 'run_pipeline' in raw_args.keys():
+          self._run_pipeline = raw_args['run_pipeline']
+
       pm.execute_notebook(
-          input_path='DSL Static Type Checking.ipynb',
+          input_path='%s.ipynb' % self._test_name,
           output_path='%s.ipynb' % self._test_name,
-          parameters={}
+          parameters=nb_params
       )
+
     else:
       subprocess.call(['dsl-compile', '--py', '%s.py' % self._test_name,
                        '--output', '%s.yaml' % self._test_name])
 
+
   def run_test(self):
     self._compile_sample()
```

Comment:
We probably need to rename the functions since compile_sample for notebook is the execution. And then the check_result function is to execute the sample. Should we separate the steps as follows:
compile(): compile dsl to pipeline, conver ipynb to py
execute(): submit pipeline or run the notebook py
check()
WDYT?

Reply:
That requires further refactoring in check_notebook_results.py and run_sample_test.py, in order to separate execute and check. Will let you know when I'm done.

Reply:
BTW, There is now a nice function that compiles the pipeline, gets experiment and submits the run. All in a single line:
```kfp.run_pipeline_func_on_cluster(automl_pipeline, arguments={})```

Reply:
Factorized into following 3 steps:
compile(): py to pipeline or notebook to py (after papermill preparation)
execute(): py retrieve config and submit for pipeline run if needed. run notebook (which usually contains pipline run)
check().

Reply:
> BTW, There is now a nice function that compiles the pipeline, gets experiment and submits the run. All in a single line:
> `kfp.run_pipeline_func_on_cluster(automl_pipeline, arguments={})`

Thanks for the info! Will do that in a following PR.